src/content-feature.js

```diff
@@ -18,33 +19,64 @@ export default class ContentFeature {
     }
 
     /**
+     * Get the value of a config setting.
+     * If the value is not set, return the default value.
+     * If the value is not an object, return the value.
+     * If the value is an object, check its type property.
      * @param {string} attrName
-     * @param {any} defaultValue
+     * @param {any} defaultValue - The default value to use if the config setting is not set
+     * @returns The value of the config setting or the default value
      */
     getFeatureAttr (attrName, defaultValue) {
-        return getFeatureAttr(this.name, this._args, attrName, defaultValue)
+        const configSetting = this.getFeatureSetting(attrName)
+        return processAttr(configSetting, defaultValue)
     }
 
     /**
      * @param {string} featureKeyName
+     * @returns {any}
      */
     getFeatureSetting (featureKeyName) {
-        return getFeatureSetting(this.name, this._args, featureKeyName)
+        let result = this._getFeatureSetting()
+        if (featureKeyName === 'domains') {
+            throw new Error('domains is a reserved feature setting key name')
+        }
+        const domainMatch = this.matchDomainFeatureSetting('domains').sort((a, b) => {
+            return a.domain.length - b.domain.length
+        })
```

Comment:
how often is `getFeatureSetting` called? just wondering if there's anything perf-wise to measure/check here? might not be, but just flagging as this is likely mutating the underlying array over and over

Reply:
Overall I think 10-15 times a page, it's probably ok. I think we could memoise the whole function response if it becomes an issue later.